src/main/java/internet/market/web/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,74 @@
+package internet.market.web.filters;
+
+import internet.market.lib.Injector;
+import internet.market.model.Role;
+import internet.market.model.User;
+import internet.market.service.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorizationFilter implements Filter {
+    private static final Injector injector = Injector.getInstance("internet.market");
+    private static final String USER_ID = "user_Id";
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+    private Map<String, Set<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/user/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/complete", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/admin/products", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/orders", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/user/orders", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/cart/product/add", Set.of(Role.RoleName.USER));
```

Comment:
Probably @vremere-a  meant this one? If it is so — it's okay cause this controller for adding product to shopping cart, but not for creating it. But he is right that it's better to add "/product/add" for admin role